src/DynamoCoreWpf/Utilities/CompactBubbleHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using Dynamo.Extensions;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.Properties;
+using ProtoCore.Mirror;
+
+namespace Dynamo.Wpf.Utilities
+{
+    /// <summary>
+    /// Helper class to process node output
+    /// </summary>
+    public static class CompactBubbleHandler
+    {
+        private static int items;
+
+        /// <summary>
+        /// Creates an instance of <cref name="CompactBubbleViewModel"/> class
+        /// from given node value
+        /// </summary>
+        /// <param name="value">Node value</param>
+        /// <returns>Instance of <cref name="CompactBubbleViewModel"/> class</returns>
+        public static CompactBubbleViewModel Process(MirrorData value)
+        {
+            items = 0;
+            var viewModel = ProcessThing(value, true);
+            viewModel.NumberOfItems = items;
+            return viewModel;
+        }        
+
+        private static CompactBubbleViewModel ProcessThing(MirrorData mirrorData, bool generateVm)
+        {
+            if (mirrorData == null) return generateVm ? new CompactBubbleViewModel(Resources.NullString, 0) : null;
```

Comment:
just a style thing, I don't know if our style guide says one way or the other, but can you put this on two lines with curly braces since it's a complex return statement